3rdParty/velocypack/src/Parser.cpp

```diff
@@ -118,6 +118,17 @@ int Parser::skipWhiteSpace(char const* err) {
   throw Exception(Exception::ParseError, err);
 }
 
+void Parser::increaseNesting() { 
+  if (++_nesting >= options->nestingLimit) {
+    throw Exception(Exception::TooDeepNesting);
+  }
```

Comment:
Here we modify the local state before throwing the exception. AFAIK the parser object can be reused, but I don't think this local state is reinitialized correctly.

Reply:
In devel, the `_nesting` attribute of the Parser is actually reset in the `parse()` method correctly, so this is not an issue there.
However, this change was missing in the backports. 
Good catch!
VelocyPack test case added via https://github.com/arangodb/velocypack/pull/122